### PR TITLE
VPA e2e test - wait for scaling fix

### LINKED ISSUE #2649: VPA - Setting up a hamster deployment flake
https://prow.k8s.io/view/gcs/kubernetes-jenkins/logs/ci-kubernetes-e2e-autoscaling-vpa-admission-controller/1204452972952555520

Test timeout waiting for a deployment even though the deployment was ready.
```
I1210 17:49:47.725] Dec 10 17:49:47.724: INFO: deployment status: v1.DeploymentStatus{ObservedGeneration:2, Replicas:3, UpdatedReplicas:3, ReadyReplicas:3, AvailableReplicas:3, UnavailableReplicas:0, Conditions:[]v1.DeploymentCondition{v1.DeploymentCondition{Type:"Progressing", Status:"True", LastUpdateTime:v1.Time{Time:time.Time{wall:0x0, ext:63711596861, loc:(*time.Location)(0x75e4d40)}}, LastTransitionTime:v1.Time{Time:time.Time{wall:0x0, ext:63711596861, loc:(*time.Location)(0x75e4d40)}}, Reason:"NewReplicaSetAvailable", Message:"ReplicaSet \"hamster-deployment-657f74bdd7\" has successfully progressed."}, v1.DeploymentCondition{Type:"Available", Status:"True", LastUpdateTime:v1.Time{Time:time.Time{wall:0x0, ext:63711596864, loc:(*time.Location)(0x75e4d40)}}, LastTransitionTime:v1.Time{Time:time.Time{wall:0x0, ext:63711596864, loc:(*time.Location)(0x75e4d40)}}, Reason:"MinimumReplicasAvailable", Message:"Deployment has minimum availability."}}, CollisionCount:(*int32)(nil)}
```

### PR DESCRIPTION
**Pull request by krzysied**
Fixes the issue when we wait for old replicas number instead the new one.

/fixes #2649

### CODE REVIEW

File: vertical-pod-autoscaler/e2e/v1/admission_controller.go [hunk 1]
```diff
@@ -559,6 +559,8 @@ func startDeploymentPods(f *framework.Framework, deployment *appsv1.Deployment)
 	gomega.Expect(err).NotTo(gomega.HaveOccurred())
 	gomega.Expect(afterScale.Spec.Replicas).To(gomega.Equal(desiredPodCount), fmt.Sprintf("expected %d replicas after scaling", desiredPodCount))
 
+	deployment, err = c.AppsV1().Deployments(ns).Get(deployment.Name, metav1.GetOptions{})
```
Comment (jbartosik) [inline vertical-pod-autoscaler/e2e/v1/admission_controller.go]: Please add a comment explaining why we do this. We had the bug for a long time and nobody noticed so I think it's not obvious

Comment (krzysied) [reply vertical-pod-autoscaler/e2e/v1/admission_controller.go]: Done.

**Pull request merged**